scripts/lintcommit.lua

```diff
@@ -0,0 +1,44 @@
+function commit_message_is_ok(commit_message)
+  if string.match(commit_message, '^ci: .*$') then
+    return true
+  end
+end
+
+function main()
+  local current_branch = vim.trim(vim.fn.system({'git', 'branch', '--show-current'}))
+  local ancestor_commit = vim.trim(vim.fn.system({'git', 'merge-base', 'master', current_branch}))
+  local commits_on_current_pr = vim.trim(vim.fn.system({'git', 'rev-list', ancestor_commit .. ".." .. current_branch}))
+
+  local all_commits = {}
+  for substring in commits_on_current_pr:gmatch("%S+") do
+     table.insert(all_commits, substring)
+  end
+
+  for index, commit_hash in ipairs(all_commits) do
+    local message = vim.trim(vim.fn.system({'git', 'show', '-s', '--format=%s' , commit_hash}))
+    if not commit_message_is_ok(message) then
+      vim.cmd("cquit 1")
+    end
+  end
+end
+
+function testing()
```

Comment:
this is a good pattern for our scripts, I would keep it around as an optional thing (in case you planned to remove it)